### DISCUSSION TITLE
Random factory reset

### DISCUSSION THREAD

**Discussion by hlovatt** [micropython/micropython#11040] [STM32 / Pyboard]
We are seeing some PyBoard-D 722 processor boards randomly, for no apparent reason, factory resetting. Has anyone seen similar? Anyone know why this happens? Definitely not instigated using usr switch and reset!

Any help greatly appreciated,

-- Howard.

**Comment by dpgeorge** [comment]
It is possible at first power up (a cold boot) for the user switch to register as pressed for a short time, due to capacitances.  This would be apparent by the on-board LED lighting up briefly during a cold boot.  But this should not last long enough to trigger the factor reset.  It only lasts long enough to select the normal boot mode.

Separately to that, I have before seen a case with a different ST MCU (STM32WB55) where a GPIO being used to select the boot mode would read incorrectly at reset.  I never got to the bottom of that issue though.

When do you see the factor reset triggering, is it only during a cold boot?

**Comment by hlovatt** [comment]
I think it is happening during a cold boot. Unfortunately site is remote and intelligence on the ground consequently limited. Will investigate the boot mode hypothesis further. Agree that capacitance across user switch an unlikely cause. -- Howard. On 16 Mar 2023, at 2:26 pm, Damien George ***@***.***> wrote:﻿
It is possible at first power up (a cold boot) for the user switch to register as pressed for a short time, due to capacitances.  This would be apparent by the on-board LED lighting up briefly during a cold boot.  But this should not last long enough to trigger the factor reset.  It only lasts long enough to select the normal boot mode.
Separately to that, I have before seen a case with a different ST MCU (STM32WB55) where a GPIO being used to select the boot mode would read incorrectly at reset.  I never got to the bottom of that issue though.
When do you see the factor reset triggering, is it only during a cold boot?
